maintainers/flake-module.nix

```diff
@@ -0,0 +1,365 @@
+{ lib, getSystem, inputs, ... }:
+
+{
+  imports = [
+    inputs.pre-commit-hooks.flakeModule
+  ];
+
+  perSystem = { lib, config, pkgs, ... }: {
+
+    # https://flake.parts/options/pre-commit-hooks-nix.html#options
+    pre-commit.settings = {
+      hooks = {
+        clang-format.enable = true;
+        nixpkgs-fmt.enable = true;
+      };
+
+      excludes = [
+        # We don't want to format test data
+        "tests/(?!nixos/).*\.nix"
+
+        # Don't format vendored code
+        "src/toml11/"
+
+        # We haven't applied formatting to pre-existing files yet
+        "doc/manual/generate-manpage.nix"
+        "doc/manual/generate-options.nix"
+        "doc/manual/redirects.js"
+        "doc/manual/theme/highlight.js"
+        "doc/manual/utils.nix"
+        "docker.nix"
+        "flake.nix"
```

Comment:
I wouldn't mind if that would be reusable and also working for editorconfig but thats probably out of scope.

Reply:
Editorconfig doesn't suffer from large diffs as much, I would think, so editorconfig won't really need this list. We could just apply its suggestions immediately.